src/config/rcxml.c

```diff
@@ -241,6 +247,74 @@ fill_region(char *nodename, char *content)
 	}
 }
 
+static void
+fill_action_query(char *nodename, char *content, struct action *action)
+{
+	string_truncate_at_pattern(nodename, ".keybind.keyboard");
+	string_truncate_at_pattern(nodename, ".mousebind.context.mouse");
+
+	if (!strcasecmp(nodename, "query.action")) {
+		current_action_query = NULL;
+	}
+
+	string_truncate_at_pattern(nodename, ".query.action");
+
+	if (!content) {
+		return;
+	}
+
+	if (!current_action_query) {
+		struct wl_list *queries = action_get_qlist(action, "query");
+		if (!queries) {
+			action_arg_add_qlist(action, "query");
+			queries = action_get_qlist(action, "query");
+		}
+		current_action_query = znew(*current_action_query);
+		wl_list_append(queries, &current_action_query->link);
+	}
+
+	if (!strcmp(nodename, "identifier")) {
+		current_action_query->identifier = xstrdup(content);
+	} else if (!strcmp(nodename, "title")) {
+		current_action_query->title = xstrdup(content);
+	}
+}
+
+static void
+fill_child_action(char *nodename, char *content, struct action *parent,
+		  const char *arg_name)
+{
+	string_truncate_at_pattern(nodename, ".keybind.keyboard");
+	string_truncate_at_pattern(nodename, ".mousebind.context.mouse");
+	string_truncate_at_pattern(nodename, ".then.action");
+	string_truncate_at_pattern(nodename, ".else.action");
+
+	// TODO: check for action type
+
+	if (!strcasecmp(nodename, "action")) {
+		current_child_action = NULL;
+	}
+
+	if (!content) {
+		return;
+	}
+
+	struct wl_list *siblings = action_get_alist(parent, arg_name);
+	if (!siblings) {
+		action_arg_add_alist(parent, arg_name);
+		siblings = action_get_alist(parent, arg_name);
+	}
+
+	if (!strcmp(nodename, "name.action")) {
+		current_child_action = action_create(content);
+		wl_list_append(siblings, &current_child_action->link);
+	} else if (!current_child_action) {
+		wlr_log(WLR_ERROR, "TODO hurr durr wrong order");
```

Comment:
This log message should maybe be a bit more formal :)
We could also look into having the same behavior everywhere when this happens.

Reply:
Yeah, done :)